sanity-io/sdk
# feat: add TypeDoc

### Description

This PR adds TypeDoc documentation and the `docs` command.

Also adds a shared `package.config` in a similar convention to the Sanity monoorepo.

https://sdk-docs-git-feat-add-typedoc.sanity.dev/
![CleanShot 2024-11-14 at 09 59 53@2x](https://github.com/user-attachments/assets/26764fd7-ff4a-49ea-8067-bc5b8fd0de06)



### What to review
Check the monorepo configuration and docs output

### Testing
1. run `pnpm run docs`
2. Check out the docs produced in the top level `docs` folder

### Fun gif
![book documentaioun](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExMXNwcW9paXo3djl4bjl2Y3JxY2VzcXd6b2Y2OGdxemw0eWdhcGRqMSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/WoWm8YzFQJg5i/giphy.gif)
